### PR TITLE
Implement str iterator support

### PR DESCRIPTION
**Pull request by mpajkowski**
Hello,

I added iteration support for ```str```.

### CODE REVIEW

File: vm/src/obj/objstr.rs [hunk 1]
```diff
@@ -1025,6 +1094,14 @@ impl PyString {
         let encoded = PyBytes::from_string(&self.value, &encoding, vm)?;
         Ok(encoded.into_pyobject(vm)?)
     }
+
+    #[pymethod(name = "__iter__")]
+    fn iter(zelf: PyRef<Self>, _vm: &VirtualMachine) -> PyStringIterator {
+        PyStringIterator {
+            position: Cell::new(0),
```
Comment (coolreader18) [inline vm/src/obj/objstr.rs]: I believe you're missing the `__reversed__`  method somewhere here?

Comment (mpajkowski) [reply vm/src/obj/objstr.rs]: Actually, I checked it using CPython (3.7.3):

```
>>> hasattr(str, "__reversed__")
False
```

Should I implement ```__reversed__``` despite this fact or leave it unchanged?

Comment (mpajkowski) [reply vm/src/obj/objstr.rs]: Thanks for your feedback :) I'll deliver corrections today.

**Comment by mpajkowski** [comment]
Thanks for your feedback :) I'll deliver corrections today.

File: vm/src/obj/objstr.rs (hunk 1 above)
Comment (coolreader18) [reply vm/src/obj/objstr.rs]: How is the `PyStringReverseIterator` used then?

Comment (mpajkowski) [reply vm/src/obj/objstr.rs]: Within ```reverse```:

```
string = "123456789"
reversed_iter = reversed(string)
```

Comment (coolreader18) [reply vm/src/obj/objstr.rs]: Right, but nowhere in your code do you construct a `PyStringReverseIterator`, so there's no way to actually get one in Python code.

Comment (mpajkowski) [reply vm/src/obj/objstr.rs]: Ok! I'll fix it in a moment.

**Comment by mpajkowski** [comment]
I corrected lint warning and added ```__reverse__``` method - it seems to be an easiest solution.

**Pull request merged**